### ISSUE TITLE
about zod customize some error messages when creating a schema.

### ISSUE THREAD

**Issue by Asensiki** [elierotenberg/fastify-zod#24]
**When I created zod schema, I customized some error messages for verification.
But when I tried to verify, the custom error message didn't work.**

### code 

`user.scheme.ts`

```js 
const loginSchema = z.object({
  email: z
    .string()
    .email({ message: "Please enter the correct email address" }),
  password: z.string(),
});

export const { schemas: userSchemas, $ref } = buildJsonSchemas({
  ....
  loginSchema,
});


```

`user.route.ts`

```js

async function userRoutes(server: FastifyInstance) {
  server.post(
    "/login",
    {
      schema: {
        body: $ref("loginSchema"),
        response: {
          200: $ref("loginResponseSchema"),
        },
      },
    },
    loginHandler
  );
}


```

`app.ts`
```js

for (const schema of [...userSchemas, ...productSchemas]) {
    server.addSchema(schema);
  }

```

### Send a request

```
{"email":"ad12dadasda","password":""}
```

### response

```json

{
  "statusCode": 400,
  "error": "Bad Request",
  "message": "body.email should match format \"email\""
}

```

**Comment by R4Y-repo** [comment]
I have the same issue and tried everything and still doesn't pick the custom error message
Did u know how to fix it?

**Comment by ali-issa** [comment]
Would be great if we get an update on this since I'm facing the same issue.

**Comment by Asensiki** [comment]
> 

No, I used another library.   👉 `fastify-type-provider-zod`

**Comment by Asensiki** [comment]
> R4Y-repo

No, I used another library.   👉 `fastify-type-provider-zod`

**Comment by elierotenberg** [comment]
Hi!

Sorry for the late reply.

I have just published 1.3.2, with support for custom error messages, as well as instructions on how to enable them:

<blockquote>
- For custom error messages, you must enable error messages when building the schemas, as well as [configuring fastify to handle them](https://www.fastify.io/docs/latest/Reference/Validation-and-Serialization/#schemaerrorformatter):

```ts
// Define custom messages
const TodoItemId = z.object({
  id: z.string().uuid("this is not a valid id!"),
});

// Then configure fastify
const f = fastify({
  ajv: {
    customOptions: {
      allErrors: true,
    },
  },
  plugins: [require("ajv-errors")],
});

await register(f, {
  jsonSchemas: buildJsonSchemas(models, { errorMessages: true }),
});
```
</blockquote>

Does this solve your problem?

**Comment by R4Y-repo** [comment]
> Hi!
> 
> Sorry for the late reply.
> 
> I have just published 1.3.2, with support for custom error messages, as well as instructions on how to enable them:
> 
> > - For custom error messages, you must enable error messages when building the schemas, as well as [configuring fastify to handle them](https://www.fastify.io/docs/latest/Reference/Validation-and-Serialization/#schemaerrorformatter):
> > ```ts
> > // Define custom messages
> > const TodoItemId = z.object({
> >   id: z.string().uuid("this is not a valid id!"),
> > });
> > 
> > // Then configure fastify
> > const f = fastify({
> >   ajv: {
> >     customOptions: {
> >       allErrors: true,
> >     },
> >   },
> >   plugins: [require("ajv-errors")],
> > });
> > 
> > await register(f, {
> >   jsonSchemas: buildJsonSchemas(models, { errorMessages: true }),
> > });
> > ```
> 
> Does this solve your problem?

what would be the right code for fastify-zod with typescript?
